### DISCUSSION TITLE
Why are there repeated reads during a write session?

### DISCUSSION THREAD

**Discussion by maxfreu** [davidfrantz/force#233] [General]
I wonder where the necessity for the repeated reads during a write arises from, specifically this https://github.com/davidfrantz/force/blob/20f18482ceaaf110dd994adde3ef4b34a60ef775/src/cross-level/brick-cl.c#L828-L891 code block.

The code comment says "read and rewrite brick (safer when using compression)". Does this mean, when any sort of compression is used, the entire brick is read, updated and rewritten during write?

Edit: To avoid the xy problem: I patched force to use square chunks instead of tilesize x 1 and noticed low write performance (I set output chunksize = input chunksize). This is accompanied by the observation that, during processing, the output tile folder size goes up, then down, then up again. Then I looked further in the `write_brick` function and found the section mentioned above and wondered whether this could be the reason for slow writes. But actually I missed out on testing the original, unpatched write performance. When the writes are happening, they are pretty fast with 300MB/s on 2HDDs in raid0, it just seems like it spends quite a lot of time outputting data while the final size is only ~12G. Thread counts are all 1 for testing. Sorry for the chaos! :D

**Comment by davidfrantz** [comment]
Hi @maxfreu,

I am not exactly sure what you are referring to exactly? What is the xy problem, and what kind of module are we actually talking about?

The specific section you tagged is only relevant for the pre-processing if we want to merge an image with an image that was already processed and is overlapping. The read is necessary to take the average between those two..

**Reply by maxfreu** [reply]
That's answer enough, thanks. Forget the rest I wrote :D
